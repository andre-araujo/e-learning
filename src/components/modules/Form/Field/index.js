import { Component } from 'react';
import {
  func,
  string,
  shape,
  object,
} from 'prop-types';

class Field extends Component {
  componentDidMount() {
    const {
      name,
      initialValue,
      context,
    } = this.props;

    if (!context.fields[name]) {
      context.setFieldValue(name, initialValue);
    }
  }

  render() {
    const {
      render,
      name,
      initialValue = '',
      context,
      ...props
    } = this.props;

    let value = context.fields[name];

    if (value === undefined) {
      value = initialValue;
    }

    return render({
      ...props,
      name,
      value,
      onChange: e => context.setFieldValue(name, e.target.value),
    });
  }
}

Field.defaultProps = {
  initialValue: '',
};

Field.propTypes = {
  render: func.isRequired,
  name: string.isRequired,
  initialValue: string,
  context: shape({
    fields: object.isRequired,
    setFieldValue: func.isRequired,
  }).isRequired,
};

export default Field;
