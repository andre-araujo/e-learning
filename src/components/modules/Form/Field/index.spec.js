import React from 'react';
import { shallow } from 'enzyme';

import Field from '.';

describe('Field component', () => {
  it('should match snapshot', () => {
    const context = {
      fields: {},
      setFieldValue: jest.fn(),
    };
    expect(shallow(<Field
      name="field"
      render={(...props) => <input {...props} />}
      context={context}
    />)).toMatchSnapshot();
  });
});
