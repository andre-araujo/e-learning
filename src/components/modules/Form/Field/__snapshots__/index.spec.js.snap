// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field component should match snapshot 1`] = `
<input
  0={
    Object {
      "name": "field",
      "onChange": [Function],
      "value": "",
    }
  }
/>
`;
